02-microarray/clustering_microarray_01_heatmap.Rmd

```diff
@@ -137,9 +270,83 @@ heatmap
 dev.off()
 ```
 
-Print session info:
+Now, let's add some annotation bars to our heatmap.
+
+## Prepare metadata for annotation
 
-```{r Print session info}
-# Print session info 
+From the accompanying [paper](https://pubmed.ncbi.nlm.nih.gov/21107323/), we know that three PLX4032-sensitive parental cell lines (M229, M238 and M249) and three derived PLX4032-resistant (r) sub-lines (M229_r5, M238_r1, and M249_r4) were treated or not treated with the RAF-selective inhibitor, PLX4032 [@Nazarian2010].
+We are going to annotate our heatmap with the variables that hold the `refinebio_cell_line` and `refinebio_treatment` data.
+
+```{r}
+# Let's prepare an annotation data frame for plotting
+annotation_df <- metadata %>%
+  # We want to select the variables that we want for annotating the heatmap
+  dplyr::select(
+    refinebio_accession_code,
+    refinebio_cell_line,
+    refinebio_treatment
+  ) %>%
+  # The `pheatmap()` function requires that the row names of our annotation object matches the column names of our dataset object
+  tibble::column_to_rownames("refinebio_accession_code")
+```
+
+### Create annotated heatmap
+
+You can create an annotated heatmap by providing our annotation object to the `annotation_col` argument of the `pheatmap()` function.
+
+```{r}
+# Create and store the annotated heatmap object
+heatmap_annotated <-
+  pheatmap(
+    df_by_var,
+    cluster_rows = TRUE,
+    cluster_cols = TRUE,
+    show_rownames = FALSE,
+    annotation_col = annotation_df,
+    main = "Annotated Heatmap",
+    colorRampPalette(c(
+      "deepskyblue",
+      "black",
+      "yellow"
+    ))(25
+    ),
+    scale = "row" # Scale values in the direction of genes (rows)
+  )
+```
+
+Now that we have annotation bars on our heatmap, we have a better idea of the cell line and treatment groups that appear to cluster together.
+
```

Comment:
I think we need one sentence here that mentions the results of this heatmap. Mainly that the samples seem to cluster by their cell lines of origin but not so much by whether or not they received the `PLX4302` treatment.